### DISCUSSION TITLE
v1.67.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#3247] [Announcements]
# 1.67.0

## Hightlights
Welcome to release `v1.67.0` of Immich. We are approaching the middle of the summer, and Immich is staying as hot as the weather with a recent appearance on the front page of Hackernews, following up with making it into the trending repositories on GitHub. This release focuses on refactoring the code bases, fixing some bugs, and adding a few improvements to existing features. Some of the key highlights include:

- Add the ability to merge similar faces
- Improve range selection behavior
- Reduce RAM usage of Machine Learning container

### Merge faces
You can now select similar faces to merge them into one. All the assets belonging to those faces will also be merged. This feature is available on the web.

[Screencast from 2023-07-09 20-23-16.webm](https://github.com/immich-app/immich/assets/27055614/6128c976-5b44-4c57-b4b3-8bf92c43f0ac)

### Improve range selection behavior
This is an improvement to the shift-select behavior of the timeline on the web.

You can now jump between random points on the timeline and select photos that aren't rendered on the screen.

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="Loading ~4000 images/videos"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

🎉 Cheers! 🎉

### Server
* fix(server): transcodes failing due to storage migration happening simultaneously by @mertalev in https://github.com/immich-app/immich/pull/3071
* fix(server): thumbnail content type not being passed to stream handle by @mertalev in https://github.com/immich-app/immich/pull/3137
* refactor(server): modularize `getFfmpegOptions` by @mertalev in https://github.com/immich-app/immich/pull/3138
* refactor(server): upload config by @jrasm91 in https://github.com/immich-app/immich/pull/3148
* fix(server): enable transcoding of audioless videos by @kasgel in https://github.com/immich-app/immich/pull/3147
* refactor(server): mime types by @jrasm91 in https://github.com/immich-app/immich/pull/3197
* feat(server): Use Testcontainers for e2e tests by @etnoy in https://github.com/immich-app/immich/pull/3202
* refactor(server): calculate asset type server side by @jrasm91 in https://github.com/immich-app/immich/pull/3200
* refactor(server): file streaming logic by @mertalev in https://github.com/immich-app/immich/pull/3103

### Web
* feat(web/server): merge faces by @alextran1502 in https://github.com/immich-app/immich/pull/3121
* fix(web): previous/next asset navigation by @brighteyed in https://github.com/immich-app/immich/pull/3107
* fix(web): Places and Things thumbnail cannot be clicked by @alextran1502 in https://github.com/immich-app/immich/pull/3133
* feat(web): add empty placeholder when no assets by @faupau03 in https://github.com/immich-app/immich/pull/3155
* feat(web): improve range selection by @brighteyed in https://github.com/immich-app/immich/pull/3193
* fix(web): download live photos video part correctly by @alextran1502 in https://github.com/immich-app/immich/pull/3230
* fix(web): remove processing key events in photo viewer window by @tranquanghuy0801 in https://github.com/immich-app/immich/pull/3238

### Mobile
* feat(mobile): adding additional languages by @alextran1502 in https://github.com/immich-app/immich/pull/3161
* chore(mobile): translations update by @alextran1502 in https://github.com/immich-app/immich/pull/3158
* feat(mobile): stop asset grid rebuilds by @fyfrey in https://github.com/immich-app/immich/pull/3226

### Machine Learning
* fix(ml): race condition when loading models by @mertalev in https://github.com/immich-app/immich/pull/3207

### CLI
* feat(cli) Add new CLI by @etnoy in https://github.com/immich-app/immich/pull/3066 (but not ready for general use yet)

### Documentation and Others
* chore(docs): add catalan translation of readme by @jagjordi in https://github.com/immich-app/immich/pull/3130
* chore(deps): bump docker/setup-buildx-action from 2.8.0 to 2.9.0 by @dependabot in https://github.com/immich-app/immich/pull/3145
* chore(docs): Updated OpenAPI docs by @mertalev in https://github.com/immich-app/immich/pull/3149
* chore(docs): Update read-only-gallery.md by @stenehall in https://github.com/immich-app/immich/pull/3191
* chore(docs): upgrade deps by @jrasm91 in https://github.com/immich-app/immich/pull/3215
* chore(docs): Fix typo by @jeverling in https://github.com/immich-app/immich/pull/3216
* chore(docs): Add mobile links to front page by @jrasm91 in https://github.com/immich-app/immich/pull/3214
* chore(dev): Set dev restart policy to unless-stopped. by @aero31aero in https://github.com/immich-app/immich/pull/3123
* chore(deps): bump docker/setup-buildx-action from 2.9.0 to 2.9.1 by @dependabot in https://github.com/immich-app/immich/pull/3236

## New Contributors
* @jagjordi made their first contribution in https://github.com/immich-app/immich/pull/3130
* @kasgel made their first contribution in https://github.com/immich-app/immich/pull/3147
* @stenehall made their first contribution in https://github.com/immich-app/immich/pull/3191
* @jeverling made their first contribution in https://github.com/immich-app/immich/pull/3216
* @tranquanghuy0801 made their first contribution in https://github.com/immich-app/immich/pull/3238

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.66.1...v1.67.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.67.0'>v1.67.0</a>.</em>

**Comment by uninvitedguest** [comment]
Having issues getting the server running after the update. Receiving `500 | getaddrinfo ENOTFOUND immich-server` when loading up `ipaddress:2283`

Went through the example docker compose and updated some pieces, namely the command for server and microservices, the typesense, redis, and postgres image references, and still had no luck.

[Docker Compose Pastebin](https://pastebin.com/uzt8w2rk)

Immich Server and Microservices logs keep repeating the same error:
```Node.js v18.16.0
node:internal/modules/cjs/loader:1078

  throw err;
  ^
Error: Cannot find module '/usr/src/app/dist/main'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

**Reply by sdm1130** [reply]
Same here...

**Comment by Godd67** [comment]
same...

**Comment by Schwabsi2000** [comment]
Jup same problem

**Comment by alextran1502** [comment]
Pushing out hotfix!! 😅
